src/main.ts

```diff
@@ -933,7 +941,19 @@ function generateService(
     maybeAddComment(info, text => (requestFn = requestFn.addJavadoc(text)));
 
     requestFn = requestFn.addParameter('request', requestType(typeMap, methodDesc));
-    requestFn = requestFn.returns(responsePromise(typeMap, methodDesc));
+
+    // Use metadata as last argument for interface only configuration
+    if (!options.outputClientImpl && !options.outputEncodeMethods && !options.outputJsonMethods && options.addGrpcMetadata) {
+      requestFn = requestFn.addParameter('metadata?', "Metadata@grpc");
```

Comment:
Make the metadata an optional type, the consumer shouldn't be forced to send it.